sa/saro.go

```diff
@@ -507,7 +507,7 @@ func (ssa *SQLStorageAuthority) CountOrders(ctx context.Context, req *sapb.Count
 // CountFQDNSets counts the total number of issuances, for a set of domains,
 // that occurred during a given window of time.
 func (ssa *SQLStorageAuthorityRO) CountFQDNSets(ctx context.Context, req *sapb.CountFQDNSetsRequest) (*sapb.Count, error) {
-	if req.WindowNS == 0 || len(req.Domains) == 0 {
+	if req.Window == nil || len(req.Domains) == 0 {
```

Comment:
This isn't quite the same check: `req.Window` could be non-nil and still result in `req.Window.AsDuration() == 0`, I believe.

Now, maybe that's okay. In this case, it would mean that the caller explicitly set the window width to be 0. It won't break anything in the function, it just virtually guarantees that the SQL query will return zero results.

So I personally think it's totally fine to leave this exactly as you have it. Just important to make sure we know exactly what we're doing here.

Reply:
I've updated `core.IsAnyNilOrZero` to account for `*durationpb.Duration` similar to [another PR](https://github.com/letsencrypt/boulder/pull/7142/files#diff-694bc9a02352f68c4a0beb8e111ddb9932c49bbaef81e88a0517c76ceb61f6c7). This case has an interesting distinction that a `time.Duration` doesn't have an `IsZero()` method, but I don't believe it particularly matters. The only locations that check `req.Window` are for ratelimits: `ssa.CountFQDNSets` and `ssa.FQDNSetTimestampsForWindow`.  

Here's where I get to a sticking point though. The [new ratelimits](https://github.com/letsencrypt/boulder/blob/19582cee4b0b92cd58bfd479ba875e7844d639c4/ratelimits/limit.go#L56-L58) package bails early if a 0 duration is configured, but the [existing ratelimit](https://github.com/letsencrypt/boulder/blob/19582cee4b0b92cd58bfd479ba875e7844d639c4/ratelimit/rate-limits.go#L233-L237) package does not.

Reply:
I think that's a simple oversight on the part of the existing rate limits. I can't think of any situation in which we'd actively want to configure a duration of 0. We just didn't have as good config checking hygiene when the original rate limits were created.